code/obj/item/clothing/suits.dm

```diff
@@ -1283,32 +1283,64 @@
 	item_state = "space-neon"
 
 /obj/item/clothing/suit/space/custom // Used for nanofabs
+	icon_state = "spacemat"
+	inhand_image_icon = "s_suit"
+	item_state = "spacemat"
 	name = "bespoke space suit"
-	desc = "A suit that protects against low pressure environments, custom-made just for you!"
-	onMaterialChanged()
-		if(src.material)
-			if(material.hasProperty("thermal"))
-				var/prot = 100 - material.getProperty("thermal")
+	desc = "A custom built suit that protects your fragile body from hard vacuum."
+	var/datum/material/renf=null
+	var/prot = 0 //tired of this being initalized in the setup code so it gets a special place now
+
+	proc/setupReinforcement(var/datum/material/R) // passes the reinforcement variable, sets up protection
+		renf = R
+		if (src.material && renf)
+			if (src.material.hasProperty("thermal"))
+				var/prot = 100 - src.material.getProperty("thermal")
 				setProperty("coldprot", prot)
 				setProperty("heatprot", round(prot/2))
 			else
 				setProperty("coldprot", 30)
 				setProperty("heatprot", 15)
 
-			if(material.hasProperty("permeable"))
-				var/prot = 100 - material.getProperty("permeable")
-				setProperty("viralprot", prot)
+			if (src.material.hasProperty("permeable"))
+				var/permprot = 100 - src.material.getProperty("permeable")
+				setProperty("viralprot", permprot)
 			else
 				setProperty("viralprot", 40)
 
-			if(material.hasProperty("density"))
-				var/prot = round(material.getProperty("density") / 13)
-				setProperty("meleeprot", prot)
+			if(src.material.hasProperty("density"))
+				prot = round(src.material.getProperty("density") / 13)// for RANGED
 				setProperty("rangedprot", (0.2 + round(prot/10, 0.1)))
 			else
-				setProperty("meleeprot", 2)
 				setProperty("rangedprot", 0.4)
 
+			if(renf.hasProperty("density"))
+				prot = round(renf.getProperty("dense") / 13)// for MELEE
+				if(prot <= 3) // keep it between 3 and 6
+					setProperty("meleeprot", 3)
+				else if(prot >= 6 )
+					setProperty("meleeprot", 6)
+				else // congrats, you're within the bounds!
+					setProperty("meleeprot", prot)
+
+				var/clunk = renf.getProperty("density") // for MOVEMENT SPEED- based off density
+				if (clunk <= 15) // lighter metals = faster
+					setProperty("space_movespeed", 0.4) // since movespeed is already initalized, no need to have final conditional
+				else if (clunk >= 40)
+					setProperty("space_movespeed", 0.7) // .1 above normal spacesuits
```

Comment:
these scales are very odd. For meleeprot, you have cutoffs at 39 and 78 density, but for movespeed you have cutoffs at 15 and 40 - ideally these should have similar or matching bounds on the scaling, not (almost) mutually exclusive ones

Reply:
also, can probably use clamp instead of elif chaining